### DISCUSSION TITLE
How to setup Roadrunner-binary workspace with the Roadrunner?

### DISCUSSION THREAD

**Discussion by litan1106** [spiral-modules/roadrunner-binary#40] [Q&A]
I want to explore this project but I didn't find any guide on how to get started it.

**Comment by wolfy-j** [comment]
Would this article help - https://roadrunner.dev/docs/intro-install ?

**Reply by litan1106** [reply]
No. I am looking for a detail guide for Roadrunner development. I know how to run app with the prebuilt binary. 

Development guide example: https://github.com/kubernetes/community/blob/master/contributors/devel/development.md

**Reply by rustatian** [reply]
Kubernetes is a fairly big tool. At the moment we don't have a similar guide for the RR because, for the RR development, you need only PHP and Go installed w/o any system dependencies or hardware/network requirenments.

**Reply by rustatian** [reply]
However, for sure, you raise a good question about `workspace` for the RR development. We will create a guide on how to create a local workspace for the RR dev, but at the moment, while we don't have that guide I'll try to put some requirements here:
1. Golang 1.16 is required because we are using some methods (especially `LoadAndDelete` from sync.Map) for the 1.16.
2. Any OS is suitable. Mac/Linux/Windows/Windows-WSL.
3. RR divided into 2 parts. Roadrunner-binary which contains only releases, plugins and CLI for the RR.
4. All plugins, workers, tests, etc located in the core RR repository https://github.com/spiral/roadrunner. If you need a custom plugin, you'll need only roadrunner-binary repo. Import your plugin here: https://github.com/spiral/roadrunner-binary/blob/master/internal/container/plugins.go and that's it. Then just build the binary (`cmd/rr/main.go`).

**Reply by rustatian** [reply]
5. Core RR library contains only our standard plugins, worker, pool, event, etc implementation. In general `roadrunner-binary` should be enough to develop your own plugin and import it to your fork of the `roadrunner-binary`.
6. There is no target in the Makefile yet to build the `roadrunner-binary` local binary, but I'll add it.

**Reply by rustatian** [reply]
I'm really sorry for the lack of documentation, but we work hard to resolve this gap.

**Reply by litan1106** [reply]
@48d90782 Thanks. This is super helpful.
